oak/server/dev/dev_oak_docker.bzl

```diff
@@ -0,0 +1,46 @@
+#
+# Copyright 2020 The Project Oak Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""Rule for generating an Oak application Docker image."""
+
+load("@io_bazel_rules_docker//container:container.bzl", "container_image")
+
+def dev_oak_docker(name, application, ports):
```

Comment:
As an aside, we should figure out what to do about this when we migrate to the Rust version of the runner. Should we keep using Bazel, or just docker build, or something else?

Reply:
Are we planning to remove Bazel completely?

Reply:
That's my question to you :) Should we keep Bazel for this? Though I imagine that we would use Cargo to build the actual binary, and also to compile to Wasm, so you would have to convince Bazel to pick up those artefacts.

Reply:
It already uses Wasm modules compiled by Cargo, but we always need to check that they are compiled beforehand, since currently Bazel doesn't automatically call Bazel.
Is this (the Docker compilation) the only thing we want to keep Bazel for?

Reply:
Right, so it seems that Bazel is not helping much here, it is essentially just a wrapper for a shell script, with no dependency analysis. I don't know if we will keep Bazel for other things, I guess if we still link C++ in the runtime, we will use Bazel for that?